File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/ValidValue.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  org.hibernate.validator.ValidatorClass
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.negocio.comum.validators.ValidRule;
import br.jus.trt8.pjecalc.negocio.comum.validators.ValidValueValidator;
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.hibernate.validator.ValidatorClass;

@Target(value={ElementType.FIELD})
@Retention(value=RetentionPolicy.RUNTIME)
@ValidatorClass(value=ValidValueValidator.class)
@Documented
public @interface ValidValue {
    public byte[] groups() default {};

    public Class<? extends ValidRule> validRule();

    public String condition() default "";

    public String[] parameters() default {};

    public int flag() default 0;

    public String message() default "{MSG0004}";
}

File: src/lib/pjecalc/core/comum/validators/valid-value.ts
/**
 * PJe-Calc v2.15.1 — ValidValue (annotation TS-adaptada)
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.ValidValue
 */
import type { ValidRule } from './valid-rule';

export interface ValidValueSpec {
  /** Construtor de ValidRule (classe ou factory). */
  validRule: new () => ValidRule;
  parameters?: string[];
  flag?: number;
  groups?: number[];
  condition?: string;
  message?: string; // default "{MSG0004}"
}

export interface ValidValue_ {
  validRule: new () => ValidRule;
  parameters: string[];
  flag: number;
  groups: number[];
  condition: string;
  message: string;
}

export function validValue(spec: ValidValueSpec): ValidValue_ {
  return {
    validRule: spec.validRule,
    parameters: spec.parameters ?? [],
    flag: spec.flag ?? 0,
    groups: spec.groups ?? [],
    condition: spec.condition ?? '',
    message: spec.message ?? '{MSG0004}',
  };
}
